core/src/main/scala/kafka/server/NoOpObserver.scala

```diff
@@ -29,10 +29,15 @@ class NoOpObserver extends Observer {
   def configure(configs: Map[String, _]): Unit = {}
 
   /**
-    * Observer the record based on the given information.
+    * Observe a request and its corresponding response.
     */
   def observe(requestContext: RequestContext, request: AbstractRequest, response: AbstractResponse): Unit = {}
 
+  /**
+    * Observe a produce request
+    */
+  def observeProduceRequest(requestContext: RequestContext, produceRequest: ProduceRequest): Unit = {}
```

Comment:
why not generalize this into observerRequest() in general? (and call the above method observeResponse()